python/ray/serve/scripts.py

```diff
@@ -297,9 +306,15 @@ def run(
         cli_logger.print(f'Deploying from config file: "{config_path}".')
 
         with open(config_path, "r") as config_file:
-            config = ServeApplicationSchema.parse_obj(yaml.safe_load(config_file))
+            config_dict = yaml.safe_load(config_file)
+            # If host or port is specified as a CLI argument, they should take priority
+            # over config values.
+            override_or_default(config_dict, "host", host, DEFAULT_HTTP_HOST)
```

Comment:
Setting the defaults (`DEFAULT_HTTP_HOST` and `DEFAULT_HTTP_PORT`) here makes it seem like you can specify `--host` or `--port` without providing a host or port. Is that possible? If not, then I don't think we need the default-handling logic here.

Reply:
Is it possible that the config file does not specify a port, and the user doesn't specify using `--host` or `--port` either? I removed the default values for `--host` and `--port` above, so the values could be `None`.

Reply:
Ah good catch, yes that's possible. I'll resolve this.